### PR TITLE
Allow users to disable registrations

### PR DESCRIPTION
**Pull request by bobbyiliev**
As per a few user requests on the DevDojo site:

> https://devdojo.com/question/auth-package-option-to-disable-registration
> https://devdojo.com/question/how-to-diable-registration-feature-using-filamentphp-and-auth-package

Implementing that functionality. When disabled if someone visits the register route they will be redirected to the `/login` route and see the following message:

image

### CODE REVIEW

File: resources/views/pages/auth/login.blade.php [hunk 1]
```diff
@@ -188,10 +188,12 @@ public function authenticate()
             </form>
 
 
-            <div class="mt-3 space-x-0.5 text-sm leading-5 text-left" style="color:{{ config('devdojo.auth.appearance.color.text') }}">
-                <span class="opacity-[47%]"> {{ config('devdojo.auth.language.login.dont_have_an_account') }} </span>
-                <x-auth::elements.text-link data-auth="register-link" href="{{ route('auth.register') }}">{{ config('devdojo.auth.language.login.sign_up') }}</x-auth::elements.text-link>
-            </div>
+            @if(config('devdojo.auth.settings.registration_enabled'))
```
Comment (tnylea) [inline resources/views/pages/auth/login.blade.php:191]: We may want to add a default value of true in this config, like so:

```php
@if(config('devdojo.auth.settings.registration_enabled', true))
```

**Comment by tnylea** [comment]
Thanks for adding the test to these. Would this also prevent people from registering using Social Auth like Google or Github?

File: resources/views/pages/auth/login.blade.php (hunk 1 above)
Comment (bobbyiliev) [reply resources/views/pages/auth/login.blade.php]: Good point, just updated that!

**Comment by bobbyiliev** [comment]
> Thanks for adding the test to these. Would this also prevent people from registering using Social Auth like Google or Github?

I've just extended the PR to include that as well!

So now this includes:
- Users can disable all registration as per popular demand
- Users can also disable email registrations and leave only social registrations if they whish to

**Comment by tnylea** [comment]
Thanks dude! This looks great.

I've added a few more options and configs. Now, you can center align the text. I've also removed the `separator` if the user has disabled email registration 😉

CleanShot 2024-12-03 at 20 51 52@2x

**Review by tnylea — APPROVED** [review]
So great! Can't wait to use our Auth package on DevDojo ;)

**Pull request merged**